### PR TITLE
fix #78636 fix #102576: allow V/H Box before HBox

### PR DESCRIPTION
**Pull request by ericfont**
HBoxes now accept a V/H Box if "dropped" on them (or if "inserted" before them). This is a supplementary fix to #78636, which previously did not allow that particular case.

There is also a separate layout issue #102576 where a VBox would not generate an implicit line break when a system contains a VBox right after a HBox in a system: Measure(s),HBox,VBox,Measure(s).  I've added a line to layoutSystem to break out of the loop when the current measure is a VBox, to ensure that the VBox will always be on its own line.

I've added test cases to handle every combination of inserting HBoxes/VBoxes/Measures before HBoxes/VBoxes.  Tests both at beginning, middle, and end of a score.

### CODE REVIEW

**Comment by ericfont** [comment]
I was trying to determine if I need to apply my changes to layoutSystem() also to layoutSystem1().  I've been looking at this diff of those two functions https://github.com/ericfont/diff-layoutSystem/commit/26047b99d273828c1b4f7be018f124f7718a5ef2#diff-00be02eba9921b8927e0730d59cb5f39 and it looks to me as though people have made changes to layoutSystem() but then haven't mirrored all the changes (that don't change state) to layoutSystem1().  I almost think those two functions should be unified together, such that whenever there is a state change to use if (!undoRedo()), so that the two functions don't diverge anymore...

File: libmscore/layout.cpp [hunk 1]
```diff
@@ -2319,6 +2320,7 @@ bool Score::layoutSystem(qreal& minWidth, qreal systemWidth, bool isFirstSystem,
             if (continueFlag
                || pbreak
                || (nt == Element::Type::VBOX || nt == Element::Type::TBOX || nt == Element::Type::FBOX)
+               || (ct == Element::Type::VBOX || ct == Element::Type::TBOX || ct == Element::Type::FBOX)
```
Comment (MarcSabatella) [inline libmscore/layout.cpp]: While this makes sense on the surface, I do think there needs to be a *lot* of testing of this change.  Perhaps there was some reason why we weren't breaking the loop right away - something we need to calculate or update later in the loop or something - that will stop working.  I don't have anything specific in mind, but I just kmow how fragile this code has proven in the past.

**Comment by MarcSabatella** [comment]
I agree it would be nice not to have two separate functions here!

**Comment by dmitrio95** [comment]
Needs actualizing and rebasing for current master, so archiving this for now.

**Pull request closed without merging**